onnxscript/ir/_external_data.py

```diff
@@ -51,3 +70,213 @@ def set_base_dir(graph: _core.Graph | _core.GraphView, base_dir: str | os.PathLi
     for tensor in _all_tensors(graph, include_attributes=True):
         if isinstance(tensor, _core.ExternalTensor):
             tensor.base_dir = base_dir
+
+
+# Loading external data
+
+
+def check_external_data_file(model: _core.Model, file_path: str | os.PathLike) -> None:
+    """
+    Check if file to which external data is to be written to is empty.
+    If file already consists of external data, load external data.
+
+    Args:
+        model: Model to be converted.
+        file_path: Location to which external data is to be stored.
+    """
+    # Check if the file is empty
+    if os.stat(file_path).st_size == 0:
+        return
+    # If file is not empty, load external data
+    with open(file_path, "r+b") as data_file:
+        for value in model.graph.initializers.values():
+            if isinstance(value.const_value, _core.ExternalTensor):
+                external_tensor = value.const_value
+                external_tensor_path = (
+                    pathlib.Path(external_tensor._base_dir) / external_tensor._path
```

Comment:
Always access the public property instead of the private one.
```suggestion
                    pathlib.Path(external_tensor.base_dir) / external_tensor._path
```